### ISSUE TITLE
import generated for a @VertxGen @ProxyGen which references another is wrong

### ISSUE THREAD

**Issue by vorburger** [vert-x3/vertx-lang-js#76]
@edewit and me believe that the import generated for a @VertxGen @ProxyGen which references another is wrong - we've run into this in https://github.com/vorburger/minecraft-storeys-maker/issues/53.

@pmlopes do you know what we mean, or would you need a simple standalone reproducer project, or (broken) PR here with a beginning of a test further illustrating the problem to be able to fix this?

**Comment by vorburger** [comment]
@pmlopes dunno if you had seen this issue (I linked it in https://github.com/vorburger/minecraft-storeys-maker/issues/53#issuecomment-395868841) - is https://github.com/vert-x3/vertx-sockjs-service-proxy/issues/1 related to this? If that (vertx-sockjs-service-proxy) is where this will be fixed, then just close this one... thank you!

**Comment by pmlopes** [comment]
I know this looks messy, but the codegen for sockjs proxies lives now on https://github.com/vert-x3/vertx-sockjs-service-proxy

**Comment by vorburger** [comment]
OK so will https://github.com/vert-x3/vertx-sockjs-service-proxy/issues/1 (evenutally) fix the problem of the  wrong import statement in TS generated for a @vertxgen @proxygen which references another we've run into this in vorburger/minecraft-storeys-maker#53 ? Then just close this issue.

**Comment by pmlopes** [comment]
The idea is that by using a definition file only one implementation is present (the js one) because it's a definition file the typescript compiler knows where to locate the js implementation and paths should work correctly

**Comment by vorburger** [comment]
@pmlopes but no, @edewit and I just looked at this, and even the JS ha the wrong import statement, so even if _you only use a definition file only one implementation is present (the js one) because it's a definition file_, the exact same problem that the TS has in import is still there in the JS require; it's:

    var CommandRegistration = require('Minecraft-js/command_registration-proxy');
 
when it should be:

    var CommandRegistration = require('./command_registration-proxy');

**Comment by pmlopes** [comment]
Yes, that is why the PR: https://github.com/vert-x3/vertx-sockjs-service-proxy/pull/2/commits/bcbd3f9f6654571b5a5653bbb6487a74e83bffc7 also adds some documentation about this case. Can you share the branch (build instructions) at https://github.com/vorburger/minecraft-storeys-maker that shows the problem so I can test with this patch to see if the files are now correctly linked
